lib/faraday/request/authorization.rb

```diff
@@ -37,7 +38,7 @@ def header_from(type, *params)
           raise ArgumentError, "Unexpected params received (got #{params.size} instead of 1)"
         else
           value = params.first
-          value = value.call if value.is_a?(Proc)
+          value = value.call if value.is_a?(Proc) || value.respond_to?(:call)
```

Comment:
The `||` is on purpose, because checking the ancestors of an object should be faster than looking up a method. This way, there should be no performance penalty on existing code bases which use a `proc` already.

Reply:
Love this attention to performance, thanks for caring and explaining!